Exporter/Exporter.php

```diff
@@ -31,35 +71,39 @@ class Exporter
      */
     public function getResponse($format, $filename, SourceIteratorInterface $source)
     {
-        switch ($format) {
-            case 'xls':
-                $writer = new XlsWriter('php://output');
-                $contentType = 'application/vnd.ms-excel';
-                break;
-            case 'xml':
-                $writer = new XmlWriter('php://output');
-                $contentType = 'text/xml';
-                break;
-            case 'json':
-                $writer = new JsonWriter('php://output');
-                $contentType = 'application/json';
-                break;
-            case 'csv':
-                $writer = new CsvWriter('php://output', ',', '"', '', true, true);
-                $contentType = 'text/csv';
-                break;
-            default:
-                throw new \RuntimeException('Invalid format');
+        if (!array_key_exists($format, $this->writers)) {
+            throw new \RuntimeException(sprintf(
+                'Invalid "%s" format, supported formats are : "%s"',
+                $format,
+                implode(', ', array_keys($this->writers))
+            ));
         }
+        $writer = $this->writers[$format];
 
         $callback = function () use ($source, $writer) {
             $handler = \Exporter\Handler::create($source, $writer);
             $handler->export();
         };
 
-        return new StreamedResponse($callback, 200, array(
-            'Content-Type' => $contentType,
+        $headers = array(
             'Content-Disposition' => sprintf('attachment; filename="%s"', $filename),
-        ));
+        );
+
+        /*
+         * NEXT_MAJOR: rely on https://github.com/sonata-project/exporter/pull/90
```

Comment:
Yes, but a minor one, right?

In this case, this is not BC break.

Sullivan SENECHAL
Le 9 juin 2016 08:31, "Grégoire Paris" <notifications@github.com> a écrit :

> In Exporter/Exporter.php
> <https://github.com/sonata-project/SonataCoreBundle/pull/270#discussion_r66389124>
> :
>
> >              'Content-Disposition' => sprintf('attachment; filename="%s"', $filename),
> > -        ));
> > +        );
> > +
> > +        /*
> > +         * NEXT_MAJOR: rely on https://github.com/sonata-project/exporter/pull/90
>
> Because it implies bumping the version constraint for exporter
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sonata-project/SonataCoreBundle/pull/270/files/f1213d87ea6afa31068fea645aed891ec594cca7#r66389124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABnqNcHV6JpTdFgo3ulw37ZjLLwFyWYNks5qJ7NBgaJpZM4InAu6>
> .
>